src/platform/Tizen/AppPreference.cpp

```diff
@@ -0,0 +1,175 @@
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    int preErr     = PREFERENCE_ERROR_NONE;
+    bool isExist   = false;
+
+    preErr = preference_is_existing(key, &isExist);
+    if (preErr != PREFERENCE_ERROR_NONE)
+    {
+        err = CHIP_ERROR_INCORRECT_STATE;
+    }
+    else if (isExist == false)
+    {
+        err = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
+    }
+
+    return err;
+}
+
+CHIP_ERROR AppPreference::CheckData(const char * key)
+{
+    return __IsKeyExist(key);
+}
+
+CHIP_ERROR AppPreference::GetData(const char * key, void * data, size_t dataSize, size_t * getDataSize, size_t offset)
+{
+    CHIP_ERROR err                = CHIP_NO_ERROR;
+    int preErr                    = PREFERENCE_ERROR_NONE;
+    char * encodedData            = NULL;
+    uint8_t * decodedData         = NULL;
+    size_t encodedDataSize        = 0;
+    size_t encodedDataPaddingSize = 0;
+    size_t decodedDataSize        = 0;
+    size_t expectedDecodedSize    = 0;
+    size_t copy_size              = 0;
+
+    VerifyOrExit(data != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+    err = __IsKeyExist(key);
+    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogProgress(DeviceLayer, "Not found data [%s]", key));
+
+    preErr = preference_get_string(key, &encodedData);
+    VerifyOrExit(preErr == PREFERENCE_ERROR_NONE, err = CHIP_ERROR_INCORRECT_STATE);
+    encodedDataSize = strlen(encodedData);
+
+    if ((encodedDataSize > 0) && (encodedData[encodedDataSize - 1] == '='))
+    {
+        encodedDataPaddingSize++;
+        if ((encodedDataSize > 1) && (encodedData[encodedDataSize - 2] == '='))
+            encodedDataPaddingSize++;
+    }
+    expectedDecodedSize = ((encodedDataSize - encodedDataPaddingSize) * 3) / 4;
+
+    decodedData = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(expectedDecodedSize));
+    VerifyOrExit(decodedData != NULL, err = CHIP_ERROR_INCORRECT_STATE);
+
+    decodedDataSize = Base64Decode(encodedData, static_cast<uint16_t>(encodedDataSize), decodedData);
+
+    copy_size = min(dataSize, decodedDataSize - offset);
+    if (getDataSize != NULL)
+    {
+        *getDataSize = copy_size;
+    }
+    memset(data, 0, dataSize);
+    memcpy(data, decodedData + offset, copy_size);
+
+    ChipLogProgress(DeviceLayer, "Get data [%s:%s]", key, (char *) data);
+
+    chip::Platform::MemoryFree(decodedData);
+
+    VerifyOrExit(dataSize >= decodedDataSize - offset, err = CHIP_ERROR_BUFFER_TOO_SMALL);
+
+exit:
+    free(encodedData);
+    return err;
+}
+
+CHIP_ERROR AppPreference::SaveData(const char * key, const uint8_t * data, size_t dataSize)
+{
+    CHIP_ERROR err             = CHIP_NO_ERROR;
+    int preErr                 = PREFERENCE_ERROR_NONE;
+    char * encodedData         = NULL;
+    size_t encodedDataSize     = 0;
+    size_t expectedEncodedSize = ((dataSize + 3) * 4) / 3;
+
+    err = __IsKeyExist(key);
+    if (err == CHIP_NO_ERROR)
+    {
+        preErr = preference_remove(key);
```

Comment:
Can't you call `RemoveData(const char * key)` here as you implement the same code in the function just a bit lower?

Reply:
Thanks for your kindly review, i updated it.